### PR TITLE
FIX: min/max username length limits weren't validated

### PR DESCRIPTION
**Pull request by xfalcox**
The custom validators introduced in e0d7cda made so we ignored the mix
and max values set on site_settings.yml. That change allowed admins to
set values outside of the range defined on the yaml file.

Related to https://meta.discourse.org/t/group-names-with-more-than-60-characters-broken/232115?u=falco

### CODE REVIEW

**Comment by tgxworld** [comment]
Are there any tests which we can add here?

**Comment by xfalcox** [comment]
> Are there any tests which we can add here?

This line will fail the app boot if someone adds a min/max with a custom validation again.

https://github.com/discourse/discourse/pull/17382/files#diff-cded4e98f332295b7937f0f4fce7bc4677ade708808b1f6c04bd324541c3824cR23

**Comment by xfalcox** [comment]
> What about the validations for min/max? I think this is the one that went uncaught and is the bug we're fixing here.

Ohh I see. Check now plz

File: spec/lib/validators/min_username_length_validator_spec.rb [hunk 1]
```diff
@@ -9,6 +9,24 @@
...
+      expect do
+        SiteSetting.min_username_length = 0
+      end.to raise_error(Discourse::InvalidParameters)
+    end
+    it "fails for values above the valid range" do
+      expect do
+        SiteSetting.min_username_length = 61
+      end.to raise_error(Discourse::InvalidParameters)
+    end
+    it "works for values below the valid range" do
```
Comment (tgxworld) [inline spec/lib/validators/min_username_length_validator_spec.rb:23]:
```suggestion
    it "works for values within the valid range" do
```

File: spec/lib/validators/max_username_length_validator_spec.rb [hunk 2]
```diff
@@ -2,13 +2,32 @@
...
+      expect do
+        SiteSetting.max_username_length = 5
+      end.to raise_error(Discourse::InvalidParameters)
+    end
+    it "fails for values above the valid range" do
+      expect do
+        SiteSetting.max_username_length = 61
+      end.to raise_error(Discourse::InvalidParameters)
+    end
+    it "works for values below the valid range" do
```
Comment (tgxworld) [inline spec/lib/validators/max_username_length_validator_spec.rb:24]:
```suggestion
    it "works for values within the valid range" do
```

**Review by tgxworld — APPROVED** [review]
Looks good :) Thank you for the changes :+1:

**Pull request merged**